tests/fixtures/1_starknet.py

```diff
@@ -24,6 +25,8 @@
 logging.getLogger("asyncio").setLevel(logging.ERROR)
 logger = logging.getLogger()
 
+sys.setrecursionlimit(10000)
```

Comment:
Tests were running successfully on my local computer, but failing in the github action because of errors involving the recursive limit. 

The recursive limit being hit in copying datastructures (in `starknet_snapshot` wasnt because of cyclical references (if they were, the error would have showed up locally), but probably because the datastructures were more complex and hitting the recursive limit.

To test this, I made this (temporary) change to the recursive limit to something high (on my local computer it is set to 1000) to see if that allowed the tests to successfully run.